### ISSUE TITLE
CreateOrUpdate: Failsending request: StatusCode=400 -- Original Error: Code="SubnetNotAssociatedWithNATGateway" Message="Subnet

### ISSUE THREAD

**Issue by SauravSrivastav** [ThorstenHans/tf-aks-with-nat-gateway#1]
creating Cluster: (Managed Cluster Name "npsdbx-cluster" / Resource Group "npsdbx"): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="SubnetNotAssociatedWithNATGateway" Message="Subnet '/subscriptions/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx/resourceGroups/npsdbx/providers/Microsoft.Network/virtualNetworks/npsbdx-aks-vnet/subnets/npsbdx-aks-snet' must have a NAT gateway associated for outbound connection."

Hi Thorsten,

I was working on AKS cluster creation using Using NAT gateway and was following your article.

But when I am using the below block I am getting the error :

  outbound_type      = "userAssignedNATGateway"
    nat_gateway_profile {
      idle_timeout_in_minutes = 4
    }
    
    Can you please guide me where I am not doing things correctly

**Comment by ThorstenHans** [comment]
@SauravSrivastav thanks for reaching out. I ran into the same issue when deploying a similar infrastructure to another azure tenant... 

The way I fixed it was by simply adding the NAT Gateway as dependency on the AKS resource.

NAT Gateway and AKS are loosly coupled here (because NAT Gateway is just attached to the underlying SubNet...). That said, Terraform tries to provision them in parallel. 

This could fail if AKS provisioning is faster than NAT Gateway provisioning because AzureRM internally verifies if there is an existing NAT Gateway associated  subnet.

Long story short, add this to your `azurerm_kubernetes_cluster` resource

```terraform
resource "azurerm_kubernetes_cluster" "myaks" {
  # omitted properties
  depends_on  = [ azurerm_nat_gateway.mynatgw ]
}
```

**Comment by ThorstenHans** [comment]
I'll close this issue, If you keep on facing issues with this, feel free to re-open it
